### PR TITLE
Remove double route JSON parsing

### PR DESCRIPTION
**Pull request by zeac**
### Description

This changes reduces route calculation time up to 30% on long routes and slow CPUs.

These changes are not applicable for the code in main branch after integration of `NavigationRoute`.

### CODE REVIEW

**Comment by zeac** [comment]
`parseNativeDirectionsAlternative` can be optimized in similar way but I'm afraid to do it without clear understanding what is going on there or tests.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt [hunk 1]
```diff
@@ -18,24 +18,19 @@ suspend fun parseDirectionsResponse(
...
-                jsonObject.getString(UUID)
-            } else {
-                null
-            }
-
-            // TODO remove after https://github.com/mapbox/navigation-sdks/issues/1229
-            if (jsonObject.has(METADATA)) {
-                onMetadata(jsonObject.getString(METADATA))
+            // TODO simplify when https://github.com/mapbox/mapbox-java/issues/1292 is finished
+            val response = DirectionsResponse.fromJson(json)
```
Comment (korshaknn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:22]: afair this method doesn't handle uuid and we need to use `DirectionsResponse.fromJson(json, options, uuid)`
cc @LukasPaczos

Comment (zeac) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:22]: `uuid` is handled manually below.

Comment (korshaknn) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:22]: @zeac yes, you take it as `val uuid = response.uuid()`
but afaik response will always have `null` uuid, if it is parsed with  `DirectionsResponse.fromJson(json)`

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt [hunk 2]
```diff
@@ -264,9 +264,7 @@ class RouterWrapperTests {
                     .toUrl(),
                 routerOrigin = Onboard,
                 message = "failed for response: ${routerResultSuccessErroneousValue.value}",
-                throwable = IllegalStateException(
-                    """route response should contain "routes" array"""
-                )
+                throwable = IllegalStateException("Null routes")
```
Comment (zeac) [inline libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:267]: @LukasPaczos is there any reason to be strict in the message?

**Review by VysotskiVadim — APPROVED** [review]
Thank you for the contribution, I like the idea 👍 

Can you please add a description to the unreleased section of changelog, something like
```
#### Bug fixes and improvements
- Fixed double direction response parsing. Directions response handling is up to 30% faster. [#5747](https://github.com/mapbox/mapbox-navigation-android/pull/5747)
```

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt (hunk 2 above)
Comment (LukasPaczos) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:267]: Not really.

**Review by LukasPaczos — APPROVED** [review]
LGTM, just missing the changelog note mentioned above.

**Pull request closed without merging**